src/glmtools.jl

```diff
@@ -419,6 +431,10 @@ function devresid(::Binomial, y, μ::Real)
     end
 end
 devresid(::Gamma, y, μ::Real) = -2 * (log(y / μ) - (y - μ) / μ)
+function devresid(::Geometric, y, μ::Real)
+    μ == 0 && return oftype(μ, NaN)
+    return 2 * (xlogy(y, y / μ) - xlogy(y + 1, (y + 1) / (μ + 1)))
+end
 devresid(::InverseGaussian, y, μ::Real) = abs2(y - μ) / (y * abs2(μ))
 function devresid(d::NegativeBinomial, y, μ::Real)
```

Comment:
Can you adapt this method to use the same pattern for consistency?

Reply:
changed `oftype(η, NaN)` to `convert(float(typeof(η)), NaN)` wherever possible in `inverselink` functions since `GLM.inverselink(GLM.IdentityLink(), 0)` is throwing an error.

Also, the `devresid` function for Negative Binomial is defined as how it is defined for Geometric.